code/__HELPERS/math/ray/ray.dm

```diff
@@ -2,7 +2,7 @@
 #define RAY_CAST_DEFAULT_MAX_DISTANCE 50
 
 //step size of a raycast, used to calculate one step by multiplying with floored direction vector
-#define RAY_CAST_STEP 0.01
+#define RAY_CAST_STEP 0.25
```

Comment:
Is that how you deal with beam lag ? Are you sure it doesn't break a few other things ?

Reply:
We've discussed about it at length with @Exxion last night and agreed that this was good enough of a value to offer good enough precision (beams shouldn't go through corners much, or in any noticeable manner) while still keeping the raycasting short enough. It breaks more thing by being at 0.01 than it'll ever break at 0.25

Reply:
0.01 is sub-pixel anyway so to begin with that was way too low.

Reply:
I'm also going to work on improving this later. Namely by completely removing the constant step size and instead making it step to the next tile boundary each time. Guarantees no missed tiles and greatly decreases the number of steps, in return for requiring some ugly math.